src/scripts/main.js

```diff
@@ -1,3 +1,388 @@
+        this.cells.push(new Cell(j, i));
+      }
+    }
+  }
+
+  getCell(x, y) {
+    return this.cells.find(cell => cell.x === x && cell.y === y);
+  }
+
+  setUIValue(cell) {
+    const fieldCellText = 'field-cell';
+    const fieldRow = `.field-row:nth-of-type(${cell.y + 1})`;
+    const fieldCell = `.${fieldCellText}:nth-of-type(${cell.x + 1})`;
+    const cellUI = document.querySelector(`${fieldRow} > ${fieldCell}`);
+    const previousValue = cellUI.textContent;
+    const newValue = cell.value;
+
+    cellUI.textContent = newValue;
+
+    const classList = cellUI.classList;
+
+    classList.remove(`${fieldCellText}--${previousValue}`);
+    classList.add(`${fieldCellText}--${newValue}`);
+  }
+
+  insertRandomNumber() {
+    const emptyCells = this.cells.filter(cell => cell.value === null);
+    const randomEmptyIndex = getRandomInteger(emptyCells.length - 1);
+    const randomEmptyCell = emptyCells[randomEmptyIndex];
+    const randomNumber = getRandomInteger(10) < 10 ? 2 : 4;
+
+    if (randomEmptyCell) {
+      randomEmptyCell.value = randomNumber;
+      this.setUIValue(randomEmptyCell);
+    }
+  }
+
+  showCells(upCell, cell) {
+    this.setUIValue(upCell);
+    this.setUIValue(cell);
+  }
+
+  addPoints(points) {
+    this.score.textContent = Number(this.score.textContent) + points;
+
+    if (points === 2048) {
+      messagWin.classList.remove(hidden);
+      this.gameWon = true;
+    }
+  }
+
+  hasNeighbourTheSameValue(cell, neighbourX, neighbourY) {
+    const neighbour = this.getCell(neighbourX, neighbourY);
+
+    return neighbour && neighbour.value === cell.value;
+  }
+
+  existsEmptyCell() {
+    const emptyCells = this.cells.filter(cell => cell.value === null);
+
+    return emptyCells.length > 0;
+  }
+
+  isMovePossible(direction) {
+    let neighbourX;
+    let neighbourY;
+
+    for (const cell of this.cells) {
+      switch (direction) {
+        case directions.up:
+          neighbourX = cell.x;
+          neighbourY = cell.y - 1;
+          break;
+        case direction.right:
+          neighbourX = cell.x + 1;
+          neighbourY = cell.y;
+          break;
+        case direction.bottom:
+          neighbourX = cell.x;
+          neighbourY = cell.y + 1;
+          break;
+        default:
+          neighbourX = cell.x - 1;
+          neighbourY = cell.y;
+      }
+
+      if (this.hasNeighbourTheSameValue(cell, neighbourX, neighbourY)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  isNextMovePossible() {
+    this.hasEmptyCells = this.existsEmptyCell();
+    this.moveUpPossible = this.isMovePossible(directions.up);
+    this.moveRightPossible = this.isMovePossible(directions.right);
+    this.moveDownPossible = this.isMovePossible(directions.down);
+    this.moveLeftPossible = this.isMovePossible(directions.left);
+
+    return this.hasEmptyCells || this.moveUpPossible || this.moveRightPossible
+      || this.moveDownPossible || this.moveLeftPossible;
+  }
+
+  moveUp() {
```

Comment:
moveUp, moveDown and moveRight, moveLeft functions are very similar, you could rather declare moveX and moveY functions that could be called with arguments modifying their behaviour or a helper function for this part:
<img width="626" alt="image" src="https://github.com/mate-academy/js_2048_game/assets/87986955/6c3de03c-80e6-4340-9f2d-bb4277e3d4c9">

Reply:
I written one function to move in all directions.